src/Compilers/CSharp/Test/Semantic/Semantics/NullCoalesceAssignmentTests.cs

```diff
@@ -186,5 +187,1060 @@ void M(int? a)
+        {
+            var source = """
+                struct S
+                {
+                    void M()
+                    {
+                        S? s = null;
+                        s ??= new(a);
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,19): error CS0103: The name 'a' does not exist in the current context
+                //         s ??= new(a);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "a").WithArguments("a").WithLocation(6, 19));
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var descendantNodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var node = descendantNodes.OfType<ImplicitObjectCreationExpressionSyntax>().Single();
+
+            var typeInfo = model.GetTypeInfo(node);
+            Assert.Equal("S", typeInfo.Type.ToTestDisplayString());
+            Assert.Equal("S", typeInfo.ConvertedType.ToTestDisplayString());
+
+            var symbolInfo = model.GetSymbolInfo(node);
+            Assert.Null(symbolInfo.Symbol);
+            Assert.Collection(symbolInfo.CandidateSymbols,
+                static c => Assert.Equal("S..ctor()", c.ToTestDisplayString()));
+
+            var assignmentNode = descendantNodes.OfType<AssignmentExpressionSyntax>().Single();
+            var assignmentTypeInfo = model.GetTypeInfo(assignmentNode);
+            Assert.Equal("S", assignmentTypeInfo.Type.ToTestDisplayString());
+            Assert.Equal("S", assignmentTypeInfo.ConvertedType.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void ErrorRecovery_ImplicitObjectCreation_NotNullableValueType()
+        {
+            var source = """
+                struct S
+                {
+                    void M()
+                    {
+                        S s = default;
+                        s ??= new(a);
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,19): error CS0103: The name 'a' does not exist in the current context
+                //         s ??= new(a);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "a").WithArguments("a").WithLocation(6, 19));
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var descendantNodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var node = descendantNodes.OfType<ImplicitObjectCreationExpressionSyntax>().Single();
+
+            var typeInfo = model.GetTypeInfo(node);
+            Assert.True(typeInfo.Type.IsErrorType());
+            Assert.True(typeInfo.ConvertedType.IsErrorType());
+
+            var symbolInfo = model.GetSymbolInfo(node);
+            Assert.Null(symbolInfo.Symbol);
```

Comment:
Do we understand why this differs from the nullable version? Is it because the whole `??=` is bogus?

Reply:
Not saying that must change, just trying to understand the why.

Reply:
If type of LHS is a reference type, we know that according to semantics of `??=` operator RHS must be of the same type. If type of LHS is a nullable value type, we know, that type of RHS must be an underlying type of that nullable. However, if LHS is a non-nullable value type we have no idea what type is RHS. In this particular case it is syntactically a constructor, but what constructor? Of `Nullable<T>` or of `T` or what? So given that we cannot answer this question, we just return empty symbol info.